arduino_fw/arduino_fw.ino

```diff
@@ -0,0 +1,184 @@
+String current_arg = "";
```

Comment:
Suggestion: Let's take this opportunity pre-merge to extract the Arduino firmware and put it in another repo.